autoload/OmniSharp/actions/typedefinition.vim

```diff
@@ -0,0 +1,111 @@
+let s:save_cpo = &cpoptions
+set cpoptions&vim
+
+" Navigate to the definition of the symbol under the cursor.
+" Optional arguments:
+" Callback: When a callback is passed in, it is called after the response is
+"           returned (synchronously or asynchronously) with the found
+"           location and a flag for whether it is in a file in the project or
+"           from the metadata. This is done instead of navigating to the found
+"           location.
+" editcommand: The command to use to open buffers, e.g. 'split', 'vsplit',
+"              'tabedit' or 'edit' (default).
+function! OmniSharp#actions#typedefinition#Find(...) abort
+  let opts = { 'editcommand': 'edit' }
+  if a:0 && type(a:1) == type('') && a:1 !=# ''
+    let opts.editcommand = a:1
+  endif
+
+  let Callback = function('s:CBGotoDefinition', [opts])
+
+  call s:StdioFind(Callback)
+endfunction
+
+function! s:StdioFind(Callback) abort
+  let opts = {
+  \ 'ResponseHandler': function('s:StdioFindRH', [a:Callback]),
+  \ 'Parameters': {
+  \   'WantMetadata': v:true,
+  \ }
+  \}
+  call OmniSharp#stdio#Request('/gototypedefinition', opts)
+endfunction
+
+function! s:StdioFindRH(Callback, response) abort
+  if !a:response.Success | return | endif
+  let body = a:response.Body
+  if type(body) == type({}) && get(body, 'Definitions', v:null) != v:null
+    let definition = body.Definitions[0]
+
+    let definition.Location.FileName = substitute(definition.Location.FileName, '?', '', 'g')
+    if g:OmniSharp_lookup_metadata
+          \ && type(definition) == type({})
+          \ && type(definition.MetadataSource) == type({})
+      let Callback = function('s:CBMetadataFind', [a:Callback])
+      call s:StdioMetadataFind(Callback, definition)
+    else
+      let location = OmniSharp#locations#ParseLocation(definition.Location)
+
+      call a:Callback(location, 0)
+    endif
+  endif
+endfunction
+
+function! s:StdioMetadataFind(Callback, definition) abort
+  let typename = substitute(a:definition.MetadataSource.TypeName, '?', '', 'g')
+  let metadataSource = {
+        \ "TypeName": typename,
+        \ "ProjectName": a:definition.MetadataSource.ProjectName,
+        \ "AssemblyName": a:definition.MetadataSource.AssemblyName,
+        \ "VersionNumber": a:definition.MetadataSource.VersionNumber
+        \ }
```

Comment:
Hi @nickspoons, is there a better way to modify the `MetadataSource`'s TypeName rather than re-creating the entire object?

Reply:
I would have thought the simplest was just like this:

```vim
  let metadataSource = a:definition.MetadataSource
  let metadataSource.TypeName = substitute(metadataSource.TypeName, '?', '', 'g')
```

But is this necessary? The MetadataSource.TypeName appears to be quite simple strings, e.g. `System.Int32`, what is the substitution for?

Reply:
It contained `?` for nullable types, such as `Microsoft.AspNetCore.Builder.WebApplicationBuilder?.cs`.

Reply:
> I would have thought the simplest was just like this:
> 
> ```viml
>   let metadataSource = a:definition.MetadataSource
>   let metadataSource.TypeName = substitute(metadataSource.TypeName, '?', '', 'g')
> ```
> 
> But is this necessary? The MetadataSource.TypeName appears to be quite simple strings, e.g. `System.Int32`, what is the substitution for?

I fixed it.